dask/array/routines.py

```diff
@@ -2049,31 +2049,22 @@ def choose(a, choices):
     return elemwise(variadic_choose, a, *choices)
 
 
-def _isnonzero_vec(v):
-    return bool(np.count_nonzero(v))
+_isnonzero_vec = np.vectorize(lambda v: bool(np.count_nonzero(v)), otypes=[bool])
```

Comment:
Why the rewrite here?

Reply:
1. This looked slightly more compact and readable IMHO
2. Best practice of not changing the type of a variable after initial assignment

Ultimately it is cosmetic.

Reply:
reverted